src/pages/search-page/SearchPage.tsx

```diff
@@ -315,86 +315,113 @@ const SearchPage = () => {
     [setCurrentLayout]
   );
 
-  // util function to get collection data given uuid
-  const fillGeometryIfMissing = useCallback(
-    async (collection: OGCCollection): Promise<OGCCollection> => {
-      if (collection.getGeometry())
-        return new Promise((resolve) => resolve(collection));
-
-      const param: SearchParameters = {
-        filter: `id='${collection.id}'`,
-        properties: "id,bbox,geometry",
-      };
-
-      try {
-        const value = await dispatch(fetchResultNoStore(param)).unwrap();
-        collection.properties = value.collections[0].properties;
-        collection.extentInt = value.collections[0].extent;
-        return collection;
-      } catch (error) {
-        console.error("Error fetching collection data:", error);
-        return collection;
-      }
-    },
-    [dispatch]
-  );
-
   // Handler for click on map
   const onClickMapPoint = useCallback(
     (uuids: Array<string>) => {
       if (uuids.length === 0) {
+        // This happens when click the empty space of map
         setSelectedUuids([]);
-        collapseAllAccordions();
+        dispatch(setExpandedItem(undefined));
       } else {
-        // This set state will store the selected uuid
+        // Since map point only store uuid in its feature, so need to fetch dataset here
+        // Clicking a map dot or a result card will only add a temporary item in bookmark list, until the bookmark icon is clicked
         setSelectedUuids(uuids);
-        // This function is exposed from PinListMenu for expanding an accordion given uuid
-        setSelectedUuid(uuids[0]);
-        // This function will fetch selected dataset and insert it to pin list
-        dispatch(fetchResultByUuidNoStore(uuids[0]))
-          .unwrap()
-          .then((res: OGCCollection) => {
-            insertItemToPinList(res);
-          });
+        dispatch(fetchAndInsertTemporary(uuids[0]));
       }
     },
     [dispatch]
   );
 
-  const onClickCard = useCallback(
+  const onClickResultCard = useCallback(
     async (item: OGCCollection | undefined) => {
       if (item) {
-        // The item set to pin list assume spatial extents is there
-        const e = await fillGeometryIfMissing(item);
-        insertItemToPinList(e);
+        dispatch(fetchAndInsertTemporary(item.id));
         setSelectedUuids([item.id]);
-        setSelectedUuid(item.id);
       }
     },
-    [fillGeometryIfMissing]
+    [dispatch]
+  );
+
+  // For checking if the uuid of an item is in bookmarkItems list
+  const checkIsBookmarked = useCallback(
```

Comment:
You do not need to define checkIsBookmarked and onClickBookmark here and pass it around, you can just define the function in reducer and remove all those passing props. Then you add a emitter inside the bookmarkListReducer and give a on off event subscribe, any change of the bookmark is done via function exposed by this reducer dispatch(addBookmark(uuid)) or dispatch(removeBookmark(uuid)), inside the add function not only you update the localstorge but then fire event to each subsbscriber via the on function so that the button can be toggle automatcially.

Reply:
Inside the reducer of the bookmark, 

1. const emitter = new Emitter(); // Inside reducer
2. const on = (string (event), handle) => emitter.on(string, handler);
3. const off = (string (event), handle) => emitter.off('xxx', handle);
4. const addBookmark = (uuid) => { save to localstore && emitter.emit('ADD_BOOKMARK', handle);
5. const removeBookmark = (uuid) => { delete localstore item && emitter.emit('REMOVE_BOOKMARK', handle);
6. const checkIsBookmark = (uuid) => { return true if bookmark set in localstorr or redex)


Once you done that, remove all the props of checkIsBookmarked and onHandleBookmarks

For component call checkIsBookemarked -> dispatch(checkIsBookmarked(uuid))